core/pubnub_fetch_history.h

```diff
@@ -0,0 +1,86 @@
+/* -*- c-file-style:"stroustrup"; indent-tabs-mode: nil -*- */
+#if !defined INC_PUBNUB_FETCH_HISTORY
+#define INC_PUBNUB_FETCH_HISTORY
+
+#include "pbcc_fetch_history.h"
+
+/** Options for fetch history. */
+struct pubnub_fetch_history_options {
+    /** The maximum number of messages to return per channel if multiple channels provided. 
+     * Has to be between 1 and 25 messages. Default is 25.
+     * If single channel is provided, maximum 100 messages. Default is 100.
+     */
+    int max_per_channel;
+    /** Direction of time traversal. Default is false, which means
+     * timeline is traversed newest to oldest. */
+    bool reverse;
+    /** If provided (not NULL), lets you select a "start date", in
+     * Timetoken format. If not provided, it will default to current
+     * time. Page through results by providing a start OR end time
+     * token. Retrieve a slice of the time line by providing both a
+     * start AND end time token. Start is "exclusive" – that is, the
+     * first item returned will be the one immediately after the start
+     * Timetoken value. Default is NULL.
+     */
+    char const* start;
+    /** If provided (not NULL), lets you select an "end date", in
+     * Timetoken format. If not provided, it will provide up to the
+     * number of messages defined in the "count" parameter. Page
+     * through results by providing a start OR end time
+     * token. Retrieve a slice of the time line by providing both a
+     * start AND end time token. End is "inclusive" – that is, if a
+     * message is associated exactly with the end Timetoken, it will
+     * be included in the result. Default is NULL.
+     */
+    char const* end;
+    /** If true to recieve metadata with each history
+     * message if any. If false, no metadata per message. Defaults to
+     * false.
+     */
+    bool include_meta;
+    /** If true to recieve message type with each history
+     * message. If false, no message type per message. Defaults to
+     * false.
+     */
+    bool include_message_type;
+    /** If true to recieve uuid with each history
+     * message. If false, no uuid per message. Defaults to
+     * false.
+     */
+    bool include_uuid;
+    /** If true to recieve message actions with each history
+     * message. If false, no message actions per message. Defaults to
+     * false.
+     */
+    bool include_message_actions;
+};
+
+/** This returns the default options for fetch history transactions.
+    It's best to always call it to initialize the
+    #pubnub_fetch_history_options, since it has several parameters.
+ */
+struct pubnub_fetch_history_options pubnub_fetch_history_defopts(void);
+
+
+/** The fetch history. It is basically the same as the
+    pubnub_history(), just adding a few options that will be sent.
```

Comment:
Consider include:
`@see pubnub_history`
Some IDE's can link the right function to this mark.

Reply:
may be next time, we can add this improvement.